File: Examen_Corto/Python/ejercicio3.py
import psycopg2

def contar_vocales(palabra):
    return sum(1 for letra in palabra if letra.lower() in 'aeiou')

palabra_ingresada = input('Ingrese una palabra: ')

numero_vocales = contar_vocales(palabra_ingresada)

resultado = f'La palabra {palabra_ingresada} tiene {numero_vocales} vocales.'
print(resultado)

# Escribir en el archivo
with open('ejercicio3.txt', 'w') as file:
    file.write(resultado)

# Almacenar el resultado en la variable 'salida'
salida = resultado

# Escribir en el archivo nuevamente para garantizar que ambos bloques de código hagan lo mismo
with open('ejercicio3.txt', 'w') as file:
    file.write(resultado)

# Conexión a la base de datos PostgreSQL
try:
    conn = psycopg2.connect(
        dbname='ExamenCorto1',
        user='postgres',
        password='gabrielgrdb',
        host='localhost',
        port='5432'
    )

    cursor = conn.cursor()

    query = "insert into salida values (%s);"
    cursor.execute(query, (salida,))

    conn.commit()

except Exception as e:
    print("Error de conexión:", e)

finally:
    if conn is not None:
        conn.close()

File: Examen_Corto/Octave/ejercicio3.m







palabra_ingresada = input('Ingrese una palabra: ', 's');

numero_vocales = contar_vocales(palabra_ingresada);

resultado = ['La palabra ' palabra_ingresada ' tiene ' num2str(numero_vocales) ' vocales.'];
disp(resultado);

fid = fopen('ejercicio3.txt', 'w');
fprintf(fid, '%s', resultado);
fclose(fid);

% Almacenar el resultado en la variable 'salida'
salida = resultado;




fid = fopen("ejercicio3.txt", "w");
fprintf(fid, "%s", salida);
fclose(fid);

% Conexión a la base de datos PostgreSQL
pkg load database %cargar el paquete

conn = pq_connect(setdbopts('dbname','ExamenCorto1','host','localhost',
'port','5432','user','postgres','password','gabrielgrdb'))


query = sprintf("insert into salida values ('%s');", salida);
N = pq_exec_params(conn, query);


